src/b-splines/constant.jl

```diff
@@ -6,48 +6,16 @@ return `A[round(Int,x)]` when interpolating
 """
 Constant
 
-"""
-`define_indices_d` for a constant b-spline calculates `ix_d = round(Int,x_d)`
-"""
-function define_indices_d(::Type{BSpline{Constant}}, d, pad)
-    symix, symx = Symbol("ix_",d), Symbol("x_",d)
-    :($symix = clamp(round(Int, $symx), first(inds_itp[$d]), last(inds_itp[$d])))
+function base_rem(::Constant, x)
+    xf = round(x)
+    δx = x - xf
+    fast_trunc(Int, xf), δx
 end
 
-"""
-`coefficients` for a constant b-spline simply sets `c_d = 1` for compatibility
-with the general b-spline framework
-"""
-function coefficients(::Type{BSpline{Constant}}, N, d)
-    sym, symx = Symbol("c_",d), Symbol("x_",d)
-    :($sym = 1)
-end
+expand_index(::Constant, xi::Number, ax::AbstractUnitRange, δx) = (xi,)
 
-"""
-`gradient_coefficients` for a constant b-spline simply sets `c_d = 0` for
-compatibility with the general b-spline framework
-"""
-function gradient_coefficients(::Type{BSpline{Constant}}, d)
-    sym, symx = Symbol("c_",d), Symbol("x_",d)
-    :($sym = 0)
-end
-
-"""
-`hessian_coefficients` for a constant b-spline simply sets `c_d = 0` for
-compatibility with the general b-spline framework
-"""
-function hessian_coefficients(::Type{BSpline{Constant}}, d)
-    sym = Symbol("c_",d)
-    :($sym = 0)
-end
+value_weights(::Constant, δx) = (oneunit(δx),)
+gradient_weights(::Constant, δx) = (zero(δx),)
+hessian_weights(::Constant, δx) = (zero(δx),)
 
-function index_gen(::Type{BSpline{Constant}}, ::Type{IT}, N::Integer, offsets...) where IT<:DimSpec{BSpline}
-    if (length(offsets) < N)
-        d = length(offsets)+1
-        sym = Symbol("c_", d)
-        return :($sym * $(index_gen(IT, N, offsets..., 0)))
-    else
-        indices = [offsetsym(offsets[d], d) for d = 1:N]
-        return :(itp.coefs[$(indices...)])
-    end
-end
+padded_axis(ax::AbstractUnitRange, ::BSpline{Constant}) = ax
```

Comment:
❗️ 

This is friggin' awesome. You are a genious. You just took the most complicated parts of this library to get right, and made them into "copy-from-wikipedia".

Have I told you how much I love the code you write?